autoemulate/experimental/emulators/gaussian_process/exact.py

```diff
@@ -226,6 +230,111 @@ def get_tune_config():
+
+        Parameters
+        ----------
+        x : TensorLike
+            Input features, expected to be a 2D tensor of shape (n_samples, n_features).
+        y : TensorLike
+            Target values, expected to be a 2D tensor of shape (n_samples, n_tasks).
+        likelihood_cls : type[MultitaskGaussianLikelihood],
+            default=MultitaskGaussianLikelihood
+            Likelihood class to use for the model. Defaults to
+            `MultitaskGaussianLikelihood`.
+        mean_module_fn : MeanModuleFn, default=constant_mean
+            Function to create the mean module.
+        covar_module_fn : CovarModuleFn, default=rbf
+            Function to create the covariance module.
+        epochs : int, default=50
+            Number of training epochs.
+        activation : type[nn.Module], default=nn.ReLU
+            Activation function to use in the model.
+        lr : float, default=2e-1
+            Learning rate for the optimizer.
+        seed : int | None, default=None
+            Random seed for reproducibility. If None, no seed is set.
+        device : DeviceLike | None, default=None
+            Device to run the model on. If None, uses the default device (usually CPU or
+            GPU).
+        """
+
+        # Init device
+        TorchDeviceMixin.__init__(self, device=device)
+
+        if seed is not None:
+            set_random_seed(seed)
+
+        # Convert to 2D tensors if needed and move to device
+        x, y = self._move_tensors_to_device(*self._convert_to_tensors(x, y))
+
+        # Initialize the mean and covariance modules
+        n_features = tuple(x.shape)[1]
+        mean_module = mean_module_fn(n_features, None)
+        covar_module = covar_module_fn(n_features, None)
+
+        # Mean and covariance modules for multitask
+        num_tasks = tuple(y.shape)[1]
+        mean_module = MultitaskMean(mean_module, num_tasks=num_tasks)
+        covar_module = MultitaskKernel(covar_module, num_tasks=num_tasks, rank=1)
+
+        # TODO: identify if initialization of task covariance factor is needed for
+        # deterministic behavior
+        # with torch.no_grad():
+        #     mean_module.task_covar_factor.fill_(1.0)  # type: ignore  # noqa: PGH003
+        #     covar_module.task_covar_factor.fill_(1.0)  # type: ignore  # noqa: PGH003
+
+        # Init likelihood
+        likelihood = likelihood_cls(num_tasks=num_tasks)
+        likelihood = likelihood.to(self.device)
+
+        # Init must be called with preprocessed data
+        gpytorch.models.ExactGP.__init__(
+            self,
+            train_inputs=x,
+            train_targets=y,
+            likelihood=likelihood,
+        )
+        self.likelihood = likelihood
+        self.mean_module = mean_module
+        self.covar_module = covar_module
+        self.epochs = epochs
+        self.lr = lr
+        self.activation = activation
+        self.to(self.device)
+
+    def forward(self, x):
```

Comment:
Are we overriding the GPExact class method because we don't have batch support? Just checking my understanding.